### ISSUE TITLE
`npm run` completion results in error `(eval):32: parse error near `elif'`

### ISSUE THREAD

**Issue by soredake** [msys2/MSYS2-packages#2396]
## Describe the issue

```
/c/Users/user/next.js canary*
❯ npm run
(eval):32: parse error near `elif'
```

Same with `source`:
```
/c/Users/user/next.js canary*
❯ source <(npm completion)
/proc/self/fd/11:9: command not found: ^M
/proc/self/fd/11:32: parse error near `elif'
```
And `eval`:
```
/c/Users/user/next.js canary*
❯ eval "$(NPM_CONFIG_UPDATE_NOTIFIER=false npm completion)"
zsh: parse error near `elif'
```

Seems like source and eval run outside of my interactive shell which results in my code in .zshrc `autoload -Uz compinit && compinit` not being loaded thus not finding compdef.

## Steps to Reproduce the Problem

1. Install windows terminal, node and msys2 from chocolatey
2. Install zsh and switch msys2 to it
3. Clone some javascript repo and cd to it
4. Try complete `npm run` 

## Additional Context: Operating System, Screenshots

- OS: Windows 10 Pro 20H2

`npm completion` output:
```
#!/bin/bash
###-begin-npm-completion-###
#
# npm command completion script
#
# Installation: npm completion >> ~/.bashrc  (or ~/.zshrc)
# Or, maybe: npm completion > /usr/local/etc/bash_completion.d/npm
#

if type complete &>/dev/null; then
  _npm_completion () {
    local words cword
    if type _get_comp_words_by_ref &>/dev/null; then
      _get_comp_words_by_ref -n = -n @ -n : -w words -i cword
    else
      cword="$COMP_CWORD"
      words=("${COMP_WORDS[@]}")
    fi

    local si="$IFS"
    IFS=$'\n' COMPREPLY=($(COMP_CWORD="$cword" \
                           COMP_LINE="$COMP_LINE" \
                           COMP_POINT="$COMP_POINT" \
                           npm completion -- "${words[@]}" \
                           2>/dev/null)) || return $?
    IFS="$si"
    if type __ltrim_colon_completions &>/dev/null; then
      __ltrim_colon_completions "${words[cword]}"
    fi
  }
  complete -o default -F _npm_completion npm
elif type compdef &>/dev/null; then
  _npm_completion() {
    local si=$IFS
    compadd -- $(COMP_CWORD=$((CURRENT-1)) \
                 COMP_LINE=$BUFFER \
                 COMP_POINT=0 \
                 npm completion -- "${words[@]}" \
                 2>/dev/null)
    IFS=$si
  }
  compdef _npm_completion npm
elif type compctl &>/dev/null; then
  _npm_completion () {
    local cword line point words si
    read -Ac words
    read -cn cword
    let cword-=1
    read -l line
    read -ln point
    si="$IFS"
    IFS=$'\n' reply=($(COMP_CWORD="$cword" \
                       COMP_LINE="$line" \
                       COMP_POINT="$point" \
                       npm completion -- "${words[@]}" \
                       2>/dev/null)) || return $?
    IFS="$si"
  }
  compctl -K _npm_completion npm
fi
###-end-npm-completion-###
```

**Comment by soredake** [comment]
No such problem with linux + zsh

**Comment by Biswa96** [comment]
Zero idea about npm & nodejs & js. Did you press Ctrl+C or V to copy-paste? Is there any hidden character in that file like UTF BOM? Does changing line ending help i.e. CRLF -> LF?

**Comment by soredake** [comment]
>Did you press Ctrl+C or V to copy-paste?

Copy-paste what?

> Is there any hidden character in that file

What file...?

**Comment by Biswa96** [comment]
Just some queries to reproduce your issue.

1. Where did you download npm from?
2. Any js sample project to reproduce the issue?
3. Does it happen with zsh only?
4. Is it possible to reproduce this issue without Microsoft Windows Terminal?

**Comment by soredake** [comment]
@Biswa96 
1. Chocolatey.
2. You don't need any repo, just type `npm run ` and press tab, or try eval/source.
3. Yes, source/eval is fine with bash.
4. Yes, with C:\tools\msys64\mingw64.exe it's reproducible too.

**Comment by soredake** [comment]
My .zshrc content if this matters https://gist.github.com/soredake/441b0c3dd719eec96e2c6b04ce9f4d85

**Comment by trey-miller** [comment]
Hello, I've encountered this issue as well, and eventually found a fix (for my case anyway).

The issue was the npm completion script, which in my case was in `AppData/Roaming/nvm/v20.13.1/node_modules/npm/lib/utils/completion.sh`, had carriage returns (`\r` characters) for some reason. 

I was in vscode searching my home folder for files with "elif" near the line in the error (in my case was line 36). When I came to this file, shellcheck noted a violation of SC1017 and suggested to remove all `\r` from the file. I did, and npm completion works again!

It's possible newer versions of npm don't have this issue, or it just came through some bad file conversion when nvm downloaded it... I don't know, but at least I know which file to check if it comes back.
